### PR TITLE
Implement MovieDaoImpl

### PR DESCRIPTION
**Pull request by Resci**

### CODE REVIEW

File: src/main/java/com/mate/dao/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert movie entity");
```
Comment (tonitrus55) [inline src/main/java/com/mate/dao/MovieDaoImpl.java]: Custom exception should be extended from RuntimeException;

File: src/main/java/com/mate/dao/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> getAllMoviesQuery = session.createQuery("from Movie", Movie.class);
+            return getAllMoviesQuery.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all movies from db");
```
Comment (tonitrus55) [inline src/main/java/com/mate/dao/MovieDaoImpl.java]: Custom exception should be extended from RuntimeException;

**Review by tonitrus55 — APPROVED** [review]

File: src/main/java/com/mate/dao/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (Sofasmile) [inline src/main/java/com/mate/dao/MovieDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

File: src/main/java/com/mate/dao/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie entity");
```
Comment (Sofasmile) [inline src/main/java/com/mate/dao/MovieDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't insert movie entity " + movie, e);
```

File: src/main/java/com/mate/dao/MovieDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> getAllMoviesQuery = session.createQuery("from Movie", Movie.class);
+            return getAllMoviesQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get all movies from db");
```
Comment (Sofasmile) [inline src/main/java/com/mate/dao/MovieDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get all movies from db", e);
```

File: src/main/java/com/mate/util/HibernateUtil.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    private HibernateUtil() {
+    }
+      
+    private static SessionFactory buildSessionFactory() {
+        try {
+            return new Configuration().configure(
+                    new File("src/main/resources/hibernate.cfg.xml")).buildSessionFactory();
+        } catch (HibernateException e) {
+            System.err.println("Initial SessionFactory creation failed." + e);
```
Comment (Sofasmile) [inline src/main/java/com/mate/util/HibernateUtil.java]: let's remove System.err.println
better use logger for this

File: src/main/resources/hibernate.cfg.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        <property name="hibernate.connection.username">root</property>
+        <property name="hibernate.connection.password">toor</property>
+        <property name="hibernate.connection.pool_size">1</property>
+        <property name="hibernate.show_sql">true</property>
+        <property name="hibernate.dialect">org.hibernate.dialect.MySQL5Dialect</property>
+        <property name="hibernate.hbm2ddl.auto">update</property>
+
+        <mapping class="com.mate.model.Movie" />
+    </session-factory>
+</hibernate-configuration>
```
Comment (Sofasmile) [inline src/main/resources/hibernate.cfg.xml]: empty line

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/mate/util/HibernateUtil.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        } catch (HibernateException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+    }
+  
+    public static SessionFactory getSessionFactory() {
+        return sessionFactory;
+    }
+  
+    public static void shutdown() {
```
Comment (RomanDubovskyi) [inline src/main/java/com/mate/util/HibernateUtil.java]: we don't use for now, let's remove it

File: src/main/resources/hibernate.cfg.xml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    <session-factory>
+        <property name="hibernate.connection.driver_class">com.mysql.cj.jdbc.Driver</property>
+        <property name="hibernate.connection.url">jdbc:mysql://localhost:3306/Cinema</property>
+        <property name="hibernate.connection.username">root</property>
+        <property name="hibernate.connection.password">toor</property>
+        <property name="hibernate.connection.pool_size">1</property>
+        <property name="hibernate.show_sql">true</property>
+        <property name="hibernate.dialect">org.hibernate.dialect.MySQL5Dialect</property>
+        <property name="hibernate.hbm2ddl.auto">update</property>
+        <mapping class="com.mate.model.Movie" />
```
Comment (RomanDubovskyi) [inline src/main/resources/hibernate.cfg.xml]:
```suggestion
        <property name="hibernate.hbm2ddl.auto">update</property>
        
        <mapping class="com.mate.model.Movie" />
```
 and add empty line at the end of the file

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**